### ISSUE TITLE
no-wrap-func - found false positive

### ISSUE THREAD

**Issue by keradus** [eslint/eslint#2734]
```js
        $("#tabs li").
            removeClass("active").
            each(function() { ... })
        ;
```

Everything is OK, but, when used space between function name and `(`

```js
        $("#tabs li").
            removeClass ("active").
            each (function() { ... })
        ;
```

it raise error:
> Wrapping non-IIFE function literals in parens is unnecessary  no-wrap-func

this is false positive

**Comment by nzakas** [comment]
Can you please share the exact prior, including line numbers?

**Comment by ilyavolodin** [comment]
I think that was supposed to be "exact message".

**Comment by michaelficarra** [comment]
I think it was supposed to be "exact error".

**Comment by nzakas** [comment]
Yikes, stupid autocorrect is killing me lately. Yes, exact error.

**Comment by mysticatea** [comment]
I guess this pattern will become valid by the next way: If the `FunctionExpression` was placed at `CallExpression.arguments[0]`, and `CallExpression.arguments.length === 1`, the `no-wrap-func` rule skips it.

**Comment by keradus** [comment]
full file:
```js
/* globals $ */
/* exported foo */

function foo () {
    "use strict";

    $("#tabs li").
        removeClass ("active").
        each (function () { alert (1); })
    ;
}

```


full output (using via gulp):
```console
[12:15:10] Starting 'js:eslint'...
[12:15:11] 'js:eslint' errored after 764 ms
[12:15:11] ESLintError in plugin 'gulp-eslint'
Message:
    Wrapping non-IIFE function literals in parens is unnecessary.
Details:
    fileName: test.js
    lineNumber: 9
[12:15:11]
test.js
  9:14  error  Wrapping non-IIFE function literals in parens is unnecessary  no-wrap-func

✖ 1 problem (1 error, 0 warnings)
```

full .eslintrc file:
```json
{
    "env":{
        "browser": true,
        "node": true
    },
    "rules": {
        // Possible Errors
        "comma-dangle": [2, "never"],
        "no-cond-assign": 2,
        "no-console": 0,
        "no-constant-condition": 0,
        "no-control-regex": 0,
        "no-debugger": 2,
        "no-dupe-args": 2,
        "no-dupe-keys": 2,
        "no-duplicate-case": 2,
        "no-empty-character-class": 2,
        "no-empty": 2,
        "no-ex-assign": 2,
        "no-extra-boolean-cast": 2,
        "no-extra-parens": 0,
        "no-extra-semi": 2,
        "no-func-assign": 2,
        "no-inner-declarations": 2,
        "no-invalid-regexp": 2,
        "no-irregular-whitespace": 2,
        "no-negated-in-lhs": 2,
        "no-obj-calls": 2,
        "no-regex-spaces": 2,
        "no-reserved-keys": 2,
        "no-sparse-arrays": 2,
        "no-unreachable": 2,
        "use-isnan": 2,
        "valid-jsdoc": 2,
        "valid-typeof": 2,

        // Best Practices
        "accessor-pairs": 0,
        "block-scoped-var": 0,
        "complexity": 0,
        "consistent-return": 0,
        "curly": 2,
        "default-case": 0,
        "dot-notation": 0,
        "dot-location": [2, "object"],
        "eqeqeq": 2,
        "guard-for-in": 2,
        "no-alert": 0,
        "no-caller": 2,
        "no-div-regex": 2,
        "no-else-return": 2,
        "no-empty-label": 2,
        "no-eq-null": 2,
        "no-eval": 2,
        "no-extend-native": 2,
        "no-extra-bind": 2,
        "no-fallthrough": 2,
        "no-floating-decimal": 2,
        "no-implied-eval": 2,
        "no-iterator": 2,
        "no-labels": 2,
        "no-lone-blocks": 2,
        "no-loop-func": 2,
        "no-multi-spaces": 0,
        "no-multi-str": 2,
        "no-native-reassign": 2,
        "no-new-func": 2,
        "no-new-wrappers": 2,
        "no-new": 2,
        "no-octal-escape": 2,
        "no-octal": 2,
        "no-param-reassign": 0,
        "no-process-env": 2,
        "no-proto": 2,
        "no-redeclare": 2,
        "no-return-assign": 2,
        "no-script-url": 2,
        "no-self-compare": 2,
        "no-sequences": 2,
        "no-throw-literal": 2,
        "no-unused-expressions": 2,
        "no-void": 2,
        "no-warning-comments": 0,
        "no-with": 2,
        "radix": 2,
        "vars-on-top": 2,
        "wrap-iife": 2,
        "yoda": 0,

        // Strict Mode
        "strict": 2,

        // Variables
        "no-catch-shadow": 2,
        "no-delete-var": 2,
        "no-label-var": 2,
        "no-shadow-restricted-names": 2,
        "no-shadow": 0,
        "no-undef-init": 2,
        "no-undef": 2,
        "no-undefined": 0,
        "no-unused-vars": 0,
        "no-use-before-define": 2,

        // Stylistic Issues
        "brace-style": [2, "stroustrup", { "allowSingleLine": true }],
        "camelcase": 0,
        "comma-spacing": 2,
        "comma-style": 0,
        "consistent-this": 0,
        "eol-last": 2,
        "func-names": 0,
        "func-style": 2,
        "indent": [2, 4, {"indentSwitchCase": true}],
        "key-spacing": 0,
        "lines-around-comment": 0,
        "linebreak-style": 2,
        "max-nested-callbacks": 0,
        "new-cap": 0,
        "new-parens": 2,
        "newline-after-var": 2,
        "no-array-constructor": 2,
        "no-continue": 0,
        "no-inline-comments": 0,
        "no-lonely-if": 2,
        "no-mixed-spaces-and-tabs": 2,
        "no-multiple-empty-lines": 2,
        "no-nested-ternary": 0,
        "no-new-object": 2,
        "no-spaced-func": 0,
        "no-ternary": 0,
        "no-trailing-spaces": 0,
        "no-underscore-dangle": 0,
        "no-unneeded-ternary": 0,
        "no-wrap-func": 2,
        "object-curly-spacing": 0,
        "one-var": 2,
        "operator-assignment": 2,
        "operator-linebreak": 2,
        "padded-blocks": 0,
        "quote-props": 0,
        "quotes": 0,
        "semi-spacing": 2,
        "semi": 2,
        "sort-vars": 0,
        "space-after-keywords": 2,
        "space-before-blocks": 2,
        "space-before-function-paren": 2,
        "space-in-brackets": 0,
        "space-in-parens": 2,
        "space-infix-ops": 2,
        "space-return-throw-case": 2,
        "space-unary-ops": 2,
        "spaced-line-comment": 0,
        "wrap-regex": 0
    }
}

```


Can I provide any more information to help solving that issue ?

**Comment by mysticatea** [comment]
If this issue accepted, I will send a PR.

**Comment by ilyavolodin** [comment]
That does look like a bug.

**Comment by michaelficarra** [comment]
This rule was recently deprecated. Do we really want to continue fixing bugs in it? Also, I believe this issue was previously brought up in https://github.com/eslint/eslint/issues/2477.

**Comment by mysticatea** [comment]
really? I hadn't known it. thank you.

**Comment by keradus** [comment]
This rule is not deprecated on http://eslint.org/docs/rules/ :(
It would be great to release that info (or fix a bug until rule will be officially deprecated)

**Comment by mysticatea** [comment]
in fact, I had fixed already, so I  can send a PR.
if we continue fixing, tell me.

**Comment by keradus** [comment]
I wonder if this bug exists in rule that replaces this one.

**Comment by nzakas** [comment]
The deprecated notice hasn't been published yet, but is in master: 2142db5d9ab7175495895d3ea7086e4a7e8f7e49

The replacement is `no-extra-parens`, we should double check that on master for this issue.

**Comment by mysticatea** [comment]
`no-extra-parens` works for this!

```
> eslint -v
v0.22.1
> echo "foo (function() {});" | eslint --stdin --reset --no-eslintrc --rule "{no-extra-parens:[2,'functions']}"
> echo "foo ((function() {}));" | eslint --stdin --reset --no-eslintrc --rule "{no-extra-parens:[2,'functions']}"

<text>
  1:6  error  Gratuitous parentheses around expression  no-extra-parens

✖ 1 problem (1 error, 0 warnings)
```

**Comment by keradus** [comment]
Confirm that `no-extra-parens` works like a charm!
Thank you all for help!
